CMakeLists.txt

```diff
@@ -1,15 +1,32 @@
-cmake_minimum_required(VERSION 3.0.0)
+# CMakeLists.txt
+# Cubos Demo project root build configuration
+
+cmake_minimum_required(VERSION 3.9.0)
 project(cubos-demo VERSION 0.1.0)
 
-set(DEMO_SOURCE
-    "src/main.cpp"
-)
+# Include CubinhosGenerate
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/lib/cubos/cmake")
+include(CubinhosGenerate)
 
+# Build demo executabe
+set(DEMO_SOURCE "src/main.cpp")
 add_executable(demo ${DEMO_SOURCE})
 
 # Dependencies
 add_subdirectory(lib/cubos)
-target_link_libraries(demo cubos)
+target_link_libraries(demo cubos-core cubos-engine)
+target_include_directories(demo PUBLIC
+                          "${CMAKE_CURRENT_BINARY_DIR}"
+                          "lib/cubos"
+                          )
+set_property(TARGET demo PROPERTY CXX_STANDARD 20)
+target_compile_features(demo PUBLIC cxx_std_20)
+
+# Assets
+target_compile_definitions(demo PUBLIC DEMO_ASSETS_FOLDER="${CMAKE_CURRENT_SOURCE_DIR}/src/assets")
+
+# Components
+cubinhos_generate(demo "${CMAKE_CURRENT_SOURCE_DIR}/src/components")
```

Comment:
We were using this previously but you can now scan the entire source directory instead of just `/components`.
With that its easier to structure the code since you can define the components next to where they actually are used.